terraso_backend/apps/soil_id/models/project_soil_settings.py

```diff
@@ -90,20 +92,63 @@ class MeasurementUnit(models.TextChoices):
     def is_custom_preset(self):
         return self.depth_interval_preset == DepthIntervalPreset.CUSTOM
 
+    @property
+    def methods(self):
+        field_names = [
+            field.name.removesuffix("_enabled")
+            for field in SoilDataDepthInterval._meta.fields
+            if field.name.endswith("_enabled")
+        ]
+        return {
+            field_name + "_enabled": getattr(self, field_name + "_required")
+            for field_name in field_names
+        }
+
     def save(self, *args, **kwargs):
         dirty_fields = self.get_dirty_fields()
         with transaction.atomic():
             result = super().save(*args, **kwargs)
             if (
                 "depth_interval_preset" in dirty_fields
-                and dirty_fields.get("depth_interval_preset") != self.depth_interval_preset
+                and (new_preset := dirty_fields.get("depth_interval_preset"))
+                != self.depth_interval_preset
             ):
                 # delete project intervals
                 ProjectDepthInterval.objects.filter(project=self).delete()
                 # delete related soil data
                 DepthDependentSoilData.delete_in_project(self.project.id)
+                # create site intervals
+                self.convert_site_intervals_to_preset(new_preset)
         return result
 
+    def convert_site_intervals_to_preset(
+        self, new_preset: Optional[DepthIntervalPreset] = None, sites: Optional[list[Site]] = None
+    ):
+        if not sites:
+            sites = Site.objects.filter(project=self.project, soil_data__isnull=False)
```

Comment:
question: we leave out sites with null soil data because we'll call this method again when the soil data is created for those sites?

trying to brainstorm cases where that might cause an issue: when we query a site that doesn't yet have soil data, a default soil data object is created for the site, which would not have the project's depth intervals. does that cause problems with how you're thinking about things?

Reply:
oh i see that you added a method to create the soil data object when a site is created. then we have this unvalidated assumption that the soil data objects will exist, which will be true for all new sites for now, but might cause issues down the line